test/dotnet-compile.Tests/CompilerTests.cs

```diff
@@ -167,6 +167,24 @@ public void DepsDependencyContextIsValidOnBuild()
                 runCommand.Execute().Should().Pass();
         }
 
+        [Fact]
+        public void CanSetOutputAssemblyName()
+        {
+            var testInstance =
+                TestAssetsManager
+                    .CreateTestInstance("LibraryWithOutputAssemblyName")
+                    .WithLockFiles();
+
+            var root = testInstance.TestRoot;
+            var outputDir = Path.Combine(root, "bin");
+            var testProject = ProjectUtils.GetProjectJson(root, "LibraryWithOutputAssemblyName");
+            var buildCommand = new BuildCommand(testProject, output: outputDir, framework: DefaultFramework);
+            var result = buildCommand.ExecuteWithCapturedOutput();
+            result.Should().Pass();
+
+            new DirectoryInfo(outputDir).Should().HaveFiles(new [] { "MyLibrary.dll" });
```

Comment:
OK. I will look into it but I would prefer to do this separately. Tools are not executable so merging this change would unblock further work on tools while I am working on executables.